.github/workflows/release.yml

```diff
@@ -21,6 +21,8 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.GH_TOKEN }}
```

Comment:
Hey checkout needs GH token ?

Reply:
Not the checkout itself. GH token is needed for the maven release plugin in order to push to the main branch.

Reply:
Then why it's not at release step ?

Reply:
I would definitely try @hzerip's option, but I would not be surprized if that GITHUB_TOKEN would be used just during `deploy` which would be useful just for deploying to Github Packages

Reply:
I'm also not sure why it's needed but if it's needed then it should be at mvn:release step.

Reply:
`maven-release-plugin` changes the version in pom.xml file and pushes it to the main branch in git repo. So we need repo to be checked out by the user that has such privileges.

Reply:
Example usage of mvn release:prepare https://github.com/hazelcast/hazelcast-cloud-common-model/blob/515fa0d088d3c41ad536777e38dcbe955da6ea91/.github/workflows/tag.yml#L38

Reply:
@hzerip this is different, we're publishing to GitHub Packages there - I think @sergeyshakhov is right

Reply:
Okay sorry for confusion :/